api/envoy/config/overload/v2alpha/overload.proto

```diff
@@ -0,0 +1,62 @@
+syntax = "proto3";
+
+package envoy.config.overload.v2alpha;
+option go_package = "v2alpha";
+
+import "google/protobuf/duration.proto";
+import "google/protobuf/struct.proto";
+
+import "validate/validate.proto";
+
+// The Overload Manager provides an extensible framework to protect Envoy instances
+// from overload of various resources (memory, cpu, file descriptors, etc)
+
+message EmptyConfig {
+}
+
+message ResourceMonitor {
+  // The name of the resource monitor to instantiate. Must match a registered
+  // resource monitor type.
+  string name = 1 [(validate.rules).string.min_bytes = 1];
+
+  // Configuration for the resource monitor being instantiated.
+  google.protobuf.Struct config = 2;
+}
+
+message ThresholdTrigger {
+  // If the resource metric (ie. pressure) is greater than or equal to this value, the trigger
+  // will fire.
+  double value = 1;
```

Comment:
Related to a previous comment, if this has a limited range, it should be documented/validated here.